File: d0ubleho/4/1912.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> number(n);
    for (int i = 0; i < n; i++) {
        cin >> number[i];
    }

    int Max = number[0]; // 최대합과 부분합을 첫 번째 숫자로 설정
    int sum = number[0];

    for (int i = 1; i < n; i++) {
        sum = max(number[i], sum + number[i]); // 다음 숫자와 현재 부분합에 다음 숫자를 더한 값 중 더 큰 값
        Max = max(Max, sum); // 최대합 업데이트
    }

    cout << Max;

    return 0;
}

File: topwon1/4/1912.py
n = int(input())
sum = list(map(int,input().split()))

dp = [0] * n # 총합이 기록된 리스트
dp[0] = sum[0]
for i in range(1,n):
    dp[i] = max(sum[i],sum[i]+dp[i-1])
    # 현재 숫자 포함하여 쭉 더해온 값과 현재 숫자만을 비교했을 때 더 큰 값을 저장한다. 
print(max(dp))